### DISCUSSION TITLE
Need help with FastLED feedback

### DISCUSSION THREAD

**Discussion by Kmanggaard** [tttapa/Control-Surface#825] [Q&A]
Hi all,

My DAW sends button LED information like this: CC number for index and CC value for color. 

Using the examples I can get feedback on note input, with the color determined by index.

How can I change the example code to listen for CC?

Also, can someone give an example of how to permute the LED index array?

**Comment by tttapa** [comment]
You can use [`CCRangeFastLED`](https://tttapa.github.io/Control-Surface-doc/Doxygen/d7/de4/group__midi-input-elements-leds.html#gaa7812b2ef3dc8db4c246c9a452db81b4).

It has a [`setLEDIndexPermuter()`](https://tttapa.github.io/Control-Surface-doc/Doxygen/d3/d98/classNoteCCKPRangeFastLED.html#a353b91dbf60c992aa7de7861136f339d) method: You pass it a function that accepts an index (zero-based) and returns the permuted index, e.g.
```cpp
uint8_t swap_led3_and_led7(uint8_t index) {
  if (index == 3)
    index = 7;
  else if (index == 7)
    index = 3;
  return index;
}

CCRangeFastLED<...> midileds {...};

void setup() {
  // ...
  midileds.setLEDIndexPermuter(swap_led3_and_led7);
}
```

**Comment by benwadub** [comment]
Hello @tttapa do you think this méthod could be used to show encoder value with ws218 led strip?
I know how to output midi date from Ableton with the script
